duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/inputscreen/autocomplete/BrowserAutoCompleteSuggestionsAdapter.kt

```diff
@@ -47,11 +62,32 @@ class BrowserAutoCompleteSuggestionsAdapter(
 ) : RecyclerView.Adapter<AutoCompleteViewHolder>() {
 
     private val deleteClickListener: (AutoCompleteSuggestion) -> Unit = {
-        suggestions = suggestions.filter { suggestion -> suggestion != it }
-        notifyItemRemoved((suggestions.indexOf(it)))
+        val suggestions = getSuggestions().filter { suggestion -> suggestion != it }
+        val newItems = createItemsWithDividers(suggestions)
+        val diffResult = calculateDiff(newItems)
+        items = newItems
+        // Animate the change
+        diffResult.dispatchUpdatesTo(this)
         autoCompleteInAppMessageDismissedListener()
     }
 
+    private fun calculateDiff(newItems: List<AutoCompleteItem>): DiffResult {
+        return DiffUtil.calculateDiff(
+            object : Callback() {
+                override fun getOldListSize() = items.size
+                override fun getNewListSize() = newItems.size
+                override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean =
+                    when {
+                        items[oldItemPosition] is Divider && newItems[newItemPosition] is Divider -> true
+                        items[oldItemPosition] is Suggestion && newItems[newItemPosition] is Suggestion ->
+                            (items[oldItemPosition] as Suggestion).value == (newItems[newItemPosition] as Suggestion).value
+                        else -> false
+                    }
+                override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int) = areItemsTheSame(oldItemPosition, newItemPosition)
```

Comment:
`areContentsTheSame` is only called if `areItemsTheSame` returns true, so if you don't need to do any other checks, then returning `true` directly here should work the same way while avoiding re-running comparison again.
```suggestion
                override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int) = true
```